modules/core/role/src/Access/ManagedRolePermissionsAccessPolicy.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\farm_role\Access;
+
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Session\AccessPolicyBase;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\Core\Session\CalculatedPermissionsItem;
+use Drupal\Core\Session\RefinableCalculatedPermissionsInterface;
+use Drupal\farm_role\ManagedRolePermissionsManagerInterface;
+
+/**
+ * Grants permissions based on the user's role and managed role permissions.
+ */
+final class ManagedRolePermissionsAccessPolicy extends AccessPolicyBase {
+
+  public function __construct(
+    protected EntityTypeManagerInterface $entityTypeManager,
+    protected ManagedRolePermissionsManagerInterface $managed_role_permissions_manager,
+  ) {}
```

Comment:
@mstenta not sure if you had heard about this, but we can start leveraging "constructor property promotion" to decrease a lot of the boilerplate required around the `__construct` method! https://www.drupal.org/project/drupal/issues/3278431

I remembered this when referencing/copying from the core [`UserRolesAccessPolicy`](https://git.drupalcode.org/project/drupal/-/blob/10.4.2/core/lib/Drupal/Core/Session/UserRolesAccessPolicy.php?ref_type=tags)

Also, comment docblocks for `__construct()` can be omitted: https://www.drupal.org/node/3365111